neurodamus/ngv.py

```diff
@@ -191,22 +234,27 @@ def _show_mcd(sec):
     #        )
 
     def set_pointers(self):
-        glut_list = self._glut_list
         c = self._cellref
-        index = 0
+        # the endfeet are not included in all as they are added later.
+        # I still do not know exactly when the pointers need to be
+        # reassigned and which ones are stale. the endfeet may be already
+        # up-to-date
+        all_secs = chain(c.all, self.endfeet)
+
+        # just a safety check
+        assert len(c.all) + len(self.endfeet) + 1 == len(self.glut_list)
 
-        for sec in c.all:
-            glut = glut_list[index]
-            index += 1
+        for glut, sec in zip(self.glut_list, all_secs):
             Nd.setpointer(glut._ref_glut, "glu2", sec(0.5).cadifus)
-        soma = c.soma[0]
-        glut = glut_list[index]
-        Nd.setpointer(glut._ref_glut, "glu2", soma(0.5).cadifus)
 
     @property
```

Comment:
why not just make `_glutlist` public?

Reply:
another case of premature opt. I was considering options added this and forgot to remove it

Reply:
done. We use `sections_glut` and `soma_glut` to record this stuff. `glut_list` is a property for compatibility reasons with metabolism